moped-editor/src/utils/activityLogFormatters/mopedProjectActivity.js

```diff
@@ -29,40 +38,83 @@ export const formatProjectActivity = (change, lookupList) => {
       changeData.old === 0 ||
       changeData.old[changedField] === 0
     ) {
-      changeDescription = `Added "${
-        lookupList[change.description[0].new]
-      }" as `;
-      changeValue = entryMap.fields[changedField].label;
-      return { changeIcon, changeDescription, changeValue };
+      return {
+        changeIcon,
+        changeText: [
+          {
+            text: `Added ${lookupList[change.description[0].new]} as `,
+            style: null,
+          },
+          {
+            text: entryMap.fields[changedField].label,
+            style: "boldText",
+          },
+        ],
+      };
     }
 
     // if the new field is null or undefined, its because something was removed
     if (!lookupList[changeData.new[changedField]]) {
-      changeDescription = `Removed ${entryMap.fields[changedField].label} `;
-      changeValue = "";
-      return { changeIcon, changeDescription, changeValue };
+      return {
+        changeIcon,
+        changeText: [
+          {
+            text: `Removed ${entryMap.fields[changedField].label} `,
+            style: null,
+          },
+        ],
+      };
     }
 
     // Changing a field, but need to use lookup table to display
-    changeDescription = `Changed ${entryMap.fields[changedField].label} to `;
-    changeValue = lookupList[changeData.new[changedField]];
-  } else {
+    return {
+      changeIcon,
+      changeText: [
+        {
+          text: `Changed ${entryMap.fields[changedField].label} to `,
+          style: null,
+        },
+        {
+          text: lookupList[changeData.new[changedField]],
+          style: "boldText",
+        },
+      ],
+    };
+  }
+  // we dont need the lookup table
+  else {
     // If the update is an object, show just the field name that was updated.
     if (typeof changeData.new[changedField] === "object") {
-      changeDescription = `Changed ${entryMap.fields[changedField].label}`;
-      changeValue = "";
-      return { changeIcon, changeDescription, changeValue };
+      return {
+        changeIcon,
+        changeText: [
+          {
+            text: `Changed ${entryMap.fields[changedField].label} `,
+            style: null,
+          },
+        ],
+      };
     }
 
     // the update can be rendered as a string
-    changeDescription = `
-          Changed ${entryMap.fields[changedField].label}
-          to `;
-    changeValue =
-      changeData.new[changedField].length > 0
+    const changeValue =
+      String(changeData.new[changedField]).length > 0
```

Comment:
Can `changeData.new[changedField]` ever return as `null`? Just wondering because apparently `String(null)` returns `'null'` (as a string) which would have a length greater than 0 here. Not sure of the result of that here. Totally a TIL for me. 🤯

Reply:
good catch! I'll add a conditional to prevent nulls and undefineds slipping through